x/leverage/keeper/keeper.go

```diff
@@ -244,7 +244,20 @@ func (k Keeper) Borrow(ctx sdk.Context, borrowerAddr sdk.AccAddress, borrow sdk.
 
 	// Determine the total amount of denom borrowed (previously borrowed + newly borrowed)
 	newBorrow := borrowed.AmountOf(borrow.Denom).Add(borrow.Amount)
-	return k.setBorrow(ctx, borrowerAddr, sdk.NewCoin(borrow.Denom, newBorrow))
+	if err := k.setBorrow(ctx, borrowerAddr, sdk.NewCoin(borrow.Denom, newBorrow)); err != nil {
+		return err
+	}
+
+	// Check MaxSupplyUtilization after transaction
+	token, err := k.GetTokenSettings(ctx, borrow.Denom)
+	if err != nil {
+		return err
+	}
+	utilization := k.SupplyUtilization(ctx, borrow.Denom)
+	if utilization.GT(token.MaxSupplyUtilization) {
+		return types.ErrMaxSupplyUtilization.Wrap(utilization.String())
+	}
+	return nil
```

Comment:
Why do we do this after `setBorrow`? 
The supply utilization needs to get the total borrow, but we could calculate that before setting the new borrowed amount

Reply:
Since we already had a function to calculate supply utilization, I was able to use it directly rather than writing similar logic (of what the supply utilization would be after a change) in the `Borrow` function body.

Reply:
we need to cache these values in the future. For now this is good.